TodoAppState.js

```diff
@@ -0,0 +1,162 @@
+import {clearFilterForm, displayTodos, closeCreateTodoDialog, toggleFilterStatusPanel} from './view-functions.js';
+import {createTodo, readTodos, updateTodo, deleteTodo} from './mock-functions.js';
+import {Todo, todoStatuses, categories, priorities} from './todo-model-functions.js';
+
+function TodoAppState() {   
+    this.todos = [];
+    this.filters = {  
+        pattern: '',
+        todoStatus: todoStatuses.DEFAULT,
+        priorityFilter: priorities.DEFAULT,
+        categoryFilter: categories.DEFAULT
+    };
+
+    this.getTodosAndDisplay = () => {
+        readTodos()
+        .then(res => res.data)
+        .then(todosReceived => {
+            this.todos = todosReceived;
+            displayTodos([...this.todos], {...this.filters});
```

Comment:
Why spreading the `this.todos` and `this.filters`?
Also, displayTodos is `view-function`, it shouldn't ideally update the arguments too